### ISSUE TITLE
After upadate to 0.59.1 unable to accept any incoming livechat. Gives error "Room Not found"

### ISSUE THREAD

**Issue by pratikvav** [RocketChat/Rocket.Chat#8597]
### Description: We have now a strange issue after the Rocket.chat was updated to 0.59.1 version by Snap. Any incoming chat when the agent tries to accept gets the error "Room Not Found" , Please see the attached screenshot.

![image](https://user-images.githubusercontent.com/15137365/31873071-7ab9add0-b7de-11e7-9a63-e1d00162269e.png)

### Server Setup Information:

* Version of Rocket.Chat Server:  0.59.1
* Operating System: Ubuntu 16.04.3 LTS
* Deployment Method(snap/docker/tar/etc): Snap
* Number of Running Instances: 1
* DB Replicaset Oplog: Enabled
* Node Version: 

### Steps to Reproduce:

1. Open chat client using a user that is already there on your system. 
2. When the agent tries to accpet the chat the issue can be seen.
3. 

### Expected behavior:
Agent should be able to accept the chat.

### Actual behavior:

Gives the error as show in the screenshot, for any users that comes.

### Relevant logs:
```
[34mI20171022-06:23:38.889(-5) Exception from sub stream-notify-room id DqjNJrFjHXTYxqs5D TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-06:23:39.169(-5) Exception from sub stream-notify-room id 4cougncNpKo3YsFtc TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-06:24:01.934(-5) Exception from sub stream-notify-room id kJj7RxZ6tGHKBza5S TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-06:24:02.563(-5) Exception from sub stream-notify-room id XAbxyzL3axrYsTe9K TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-06:24:07.547(-5) Exception from sub stream-notify-room id MHQXr6T3bKAXDsXcv TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-06:24:40.004(-5) Exception from sub stream-notify-room id SHNFwCdHZMChWknbM TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-06:26:21.422(-5) Exception from sub stream-notify-room id kedEAfa5uzhbAMMjM TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-06:26:44.948(-5) Exception from sub stream-notify-room id WGDBHXTyWmzvMuXhu TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-06:26:46.963(-5) Exception from sub stream-notify-room id bEM67t5yYD7uTevM8 TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-06:26:48.816(-5) Exception from sub stream-notify-room id iWdyQT6qXMqrDtyDZ TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-06:27:39.925(-5) Exception from sub stream-notify-room id B2cAaNjKM5tYkRGQD TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-23:07:16.221(-5) Exception in defer callback: TypeError: Cannot read property 'length' of undefined   at /snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:2750:53   at /snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:320:25   at Array.reduce (native)   at Object.RocketChat.callbacks.run (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:313:6)   at /snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4115:32   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at packages/meteor.js:452:25   at runWithEnvironment (packages/meteor.js:1188:24)  
[34mI20171022-23:07:48.283(-5) Exception in defer callback: TypeError: Cannot read property 'length' of undefined   at /snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:2750:53   at /snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:320:25   at Array.reduce (native)   at Object.RocketChat.callbacks.run (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:313:6)   at /snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4115:32   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at packages/meteor.js:452:25   at runWithEnvironment (packages/meteor.js:1188:24)  
[34mI20171022-23:43:28.883(-5) Exception from sub stream-notify-room id fApBTTxqDXcr9eWTb TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-23:43:29.164(-5) Exception from sub stream-notify-room id J7neyzTBmjYntTLDG TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-23:45:10.823(-5) Exception from sub stream-notify-room id LGi5q2Lv3e6hyTgsf TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-23:45:28.235(-5) Exception from sub stream-notify-room id gWtBAm3hDvQNzeAND TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-23:45:28.886(-5) Exception from sub stream-notify-room id jT2DA7hf5Tt2aTAcz TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-23:45:51.959(-5) Exception while invoking method 'livechat:transfer' TypeError: Cannot read property 'indexOf' of undefined   at [object Object].Meteor.methods.livechat:transfer (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_livechat.js:1384:21)   at [object Object].methodsMap.(anonymous function) (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1276:26)   at [object Object].methodMap.(anonymous function) (packages/rocketchat_monitoring.js:2731:30)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:903:20   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:902:47   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:901:46   at [object Object]._.extend.protocol_handlers.method (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:874:21)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171022-23:45:58.173(-5) Exception while invoking method 'livechat:transfer' TypeError: Cannot read property 'indexOf' of undefined   at [object Object].Meteor.methods.livechat:transfer (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_livechat.js:1384:21)   at [object Object].methodsMap.(anonymous function) (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1276:26)   at [object Object].methodMap.(anonymous function) (packages/rocketchat_monitoring.js:2731:30)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:903:20   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:902:47   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:901:46   at [object Object]._.extend.protocol_handlers.method (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:874:21)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171023-00:02:10.709(-5) Exception from sub stream-notify-room id RhheW7DvZqMDDQFSM TypeError: Cannot read property 'indexOf' of undefined   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4937:25)   at Streamer.isReadAllowed (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:341:38)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_streamer.js:476:16)   at [object Object].<anonymous> (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:1295:15)   at packages/matb33_collection-hooks.js:307:21   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._handler (packages/matb33_collection-hooks.js:306:28)   at maybeAuditArgumentChecks (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1857:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1180:16   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at [object Object]._.extend._runHandler (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1179:51)   at [object Object]._.extend._startSubscription (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:1027:9)   at [object Object]._.extend.protocol_handlers.sub (/snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:815:12)   at /snap/rocketchat-server/1172/programs/server/packages/ddp-server.js:753:85  
[34mI20171023-00:12:38.679(-5) Exception in defer callback: TypeError: Cannot read property 'length' of undefined   at /snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:2750:53   at /snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:320:25   at Array.reduce (native)   at Object.RocketChat.callbacks.run (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:313:6)   at /snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4115:32   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at packages/meteor.js:452:25   at runWithEnvironment (packages/meteor.js:1188:24)  
[34mI20171023-00:13:58.796(-5) Exception in defer callback: TypeError: Cannot read property 'length' of undefined   at /snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:2750:53   at /snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:320:25   at Array.reduce (native)   at Object.RocketChat.callbacks.run (/snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:313:6)   at /snap/rocketchat-server/1172/programs/server/packages/rocketchat_lib.js:4115:32   at [object Object].EVp.withValue (packages/meteor.js:1134:15)   at packages/meteor.js:452:25   at runWithEnvironment (packages/meteor.js:1188:24)   
```

**Comment by joserribeiro26** [comment]
I think that this Issue can be merge with #8557

**Comment by sampaiodiego** [comment]
fixed by #8577
